astroquery/jplspec/core.py

```diff
@@ -0,0 +1,246 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+import os
+import warnings
+
+import astropy.units as u
+from astropy.io import ascii
+from ..query import BaseQuery
+from ..utils import async_to_sync
+# import configurable items declared in __init__.py
+from . import conf
+from . import lookup_table
+
+
+__all__ = ['JPLSpec', 'JPLSpecClass']
+
+
+def data_path(filename):
+    data_dir = os.path.join(os.path.dirname(__file__), 'data')
+    return os.path.join(data_dir, filename)
+
+
+@async_to_sync
+class JPLSpecClass(BaseQuery):
+
+    # use the Configuration Items imported from __init__.py
+    URL = conf.server
+    TIMEOUT = conf.timeout
+
+    def query_lines_async(self, min_frequency, max_frequency,
+                          min_strength=-500,
+                          max_lines=2000, molecule='All', flags=0,
+                          parse_name_locally=False,
+                          get_query_payload=False, cache=True):
+        """
+        Creates an HTTP POST request based on the desired parameters and
+        returns a response.
+
+        Parameters
+        ----------
+        min_frequency : `astropy.units`
+            Minimum frequency (or any spectral() equivalent)
+        max_frequency : `astropy.units`
+            Maximum frequency (or any spectral() equivalent)
+        min_strength : int, optional
+            Minimum strength in catalog units
+        max_lines :  int, optional
+            Maximum number of lines to query, the default is 1000.
+            The most the query allows is 100000
+
+        molecule : list, string of regex if parse_name_locally=True, optional
+            Identifiers of the molecules to search for. If this parameter
+            is not provided the search will match any species.
+
+        flags : int, optional
+            Regular expression flags.
+
+        parse_name_locally : bool, optional
+            When set to True it allows the method to parse through catdir.cat
+            in order to match the regex inputted in the molecule parameter
+            and request the corresponding tags of the matches instead.
+
+        get_query_payload : bool, optional
+            When set to `True` the method should return the HTTP request
+            parameters as a dict.
+
+        Returns
+        -------
+        response : `requests.Response`
+            The HTTP response returned from the service.
+
+        Examples
+        --------
+        >>> table = JPLSpec.query_lines(min_frequency=100*u.GHz,
+        ...                             max_frequency=200*u.GHz,
+        ...                             min_strength=-500, molecule=18003) # doctest: +SKIP
```

Comment:
could you maybe try and run this as a remote doctest? Rather than `+SKIP` it's called `+REMOTE_DATA`